File: pinyougou-parent/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;

import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * @Auther: invokersun
 * @Date: 2019/1/17   18:39
 * @Desription:
 */
@RestController
@RequestMapping("brand")
public class BrandController {
    @Reference
    private BrandService brandService;

    @RequestMapping("findAll")
    public List<TbBrand> findAll() {
        return brandService.findAll();
    }

    @RequestMapping("pagingQuery")
    public PageResult pagingQuery(@RequestParam(name = "pageNum",required = true,defaultValue = "1") int num,@RequestParam(name = "pageSize",required = true,defaultValue = "10")int size) {
        return brandService.pagingQuery(num,size);
    }
    @RequestMapping("addBrand")
    public Result addBrand(@RequestBody TbBrand tbBrand){
       return brandService.addBrand(tbBrand);
    }
    @RequestMapping("findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }
    @RequestMapping("updateBrand")
    public Result updateBrand(@RequestBody TbBrand tbBrand){
        try {
            brandService.updateBrand(tbBrand);
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败！");
        }
        return new Result(true,"修改成功!");
    }
    //删除选中项目
    @RequestMapping("deleteSelected")
    public Result deleteSelected(Long[] ids){
        try {
            brandService.deleteSlected(ids);
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败！");
        }
        return new Result(true,"删除成功!");
    }
    //条件查询
    @RequestMapping("criteriaQuery")
    public PageResult criteriaQuery(@RequestBody TbBrand tbBrand,int pageNum,int pageSize){
        return brandService.criteriaQuery(tbBrand,pageNum,pageSize);
    }
}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController', function ($scope, $http, $controller, brandService) {


    $controller('baseController',{$scope:$scope});//继承


    $scope.findAll = function () {
        $http.get("../brand/findAll").success(function (response) {
            $scope.list = response;
        })
    };
    //初始化searchEntity
    $scope.searchEntity={};

    //分页查询
    $scope.pagingQuery = function (page, size) {
        brandService.pagingQuery(page, size).success(function (response) {
            $scope.list = response.list;
            $scope.paginationConf.totalItems = response.totals;
        })
    };
    //保存brand
    $scope.saveBrand = function () {
        var object = null;
        //通过id 是否为null 判断是修改还是新建 品牌
        if ($scope.entity.id != null) {
            object = brandService.updateBrand($scope.entity);
        } else {
            object = brandService.addBrand($scope.entity);
        }
        object.success(function (response) {
            if (response.success) {
                $scope.reloadList();//重新加载
            } else {
                alert(response.resultMessage);
            }
        })
    };
    //将对应的品牌绑定到$scope.entity
    $scope.findOne = function (pojo) {
        $scope.entity = pojo;
    };

    //删除选中项
    $scope.deleteSelected = function () {
        brandService.deleteSelected($scope.selectIds).success(function (response) {
            if (response.success) {
                $scope.reloadList();//重新加载
            } else {
                alert(response.resultMessage);
            }
        });
        $scope.selectIds=[];
    };

    //条件查询
    $scope.criteriaQuery = function (page, size) {
        brandService.criteriaQuery(page, size, $scope.searchEntity).success(function (response) {
            $scope.list = response.list;
            $scope.paginationConf.totalItems = response.totals;
        })
    };
    //清空entity
    $scope.clearEntity = function () {
        $scope.entity = {};
    }
});